File: NodeJS/automata.js
const crypto = require('crypto');

function sha256(text) {
  return crypto.createHash('sha256').update(text).digest('hex');
}

function generarAutomata(n, k) {
  const nodos = [];
  const startTime = Date.now();

  for (let i = 0; i < n; i++) {
    let partida;
    if (i === 0) {
      partida = sha256(new Date().toISOString());
    } else {
      partida = nodos[i - 1].firma;
    }

    const cuerpo = Array.from({ length: k }, () => Math.floor(Math.random() * 100000) + 1);
    const data = partida + ' ' + cuerpo.join(' ');
    const firma = sha256(data);

    nodos.push({ partida, cuerpo, firma });
  }

  const endTime = Date.now();

  nodos.forEach((nodo, i) => {
    console.log(`Nodo ${i + 1}:`);
    console.log(`  Partida: ${nodo.partida}`);
    console.log(`  Cuerpo : [${nodo.cuerpo.join(', ')}]`);
    console.log(`  Firma  : ${nodo.firma}\n`);
  });

  console.log(`Tiempo de ejecución: ${(endTime - startTime)} ms`);
}

// Cambia estos valores para probar otros escenarios///////////////////////////////////////
generarAutomata(10, 200);

File: Python/Automata.py
import hashlib
import random
import time
from datetime import datetime

def sha256_hash(data: str) -> str:
    return hashlib.sha256(data.encode()).hexdigest()

def generar_nodos(n, k):
    nodos = []
    inicio = time.time()

    for i in range(n):
        if i == 0:
            partida = sha256_hash(datetime.now().isoformat())
        else:
            partida = nodos[i-1]["firma"]
        
        cuerpo = [random.randint(1, 100000) for _ in range(k)]
        datos_para_firma = partida + " " + " ".join(map(str, cuerpo))
        firma = sha256_hash(datos_para_firma)

        nodo = {
            "partida": partida,
            "cuerpo": cuerpo,
            "firma": firma
        }
        nodos.append(nodo)

    fin = time.time()
    for i, nodo in enumerate(nodos):
        print(f"Nodo {i+1}:")
        print(f"  Partida: {nodo['partida']}")
        print(f"  Cuerpo: {nodo['cuerpo']}")
        print(f"  Firma : {nodo['firma']}\n")

    print(f"Tiempo de ejecución: {fin - inicio:.6f} segundos")

# CAMBIAR VALORES AQUIIIII
generar_nodos(n=3, k=4)
